app/models/piece.rb

```diff
@@ -1,3 +1,130 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  # anywhere it say y_coord was self.y & x_coord was self.x
+
+                # Need function to automoatically keep up with current pieces coords at all times.
+  def current_pieces_coords    # Will need to be removed later after new function made.  
```

Comment:
Nice, can you think about a one-liner producing this from the real pieces?

Reply:
It's not a one-liner but one idea I had:
```
current_pieces_coords = Array.new
self.game.pieces.each do |piece|
  current_pieces_coords << [piece.x_coord, piece.y_coord]
end
return current_pieces_coords
```

Reply:
Awesome @bhk1019 - what if you use `.map`?

Reply:
Ah like:
```
self.game.pieces.map { |piece| [piece.x_coord, piece.y_coord] }
```
?

Reply:
👍

Reply:
Should the current_pieces_coords method stay in piece.rb? I only put this in here to run test. @ilyakrasnov